Jenkinsfile

```diff
@@ -117,22 +117,29 @@ node('rhel8'){
 		stage "Publish to Marketplaces"
 		unstash 'vsix'
 		def vsix = findFiles(glob: '**.vsix')
+		def platformVsixes = findFiles(glob: '**.vsix', excludes: vsix[0].path)
+
 		// VS Code Marketplace
-		withCredentials([[$class: 'StringBinding', credentialsId: 'vscode_java_marketplace', variable: 'TOKEN']]) {
-			def platformVsixes = findFiles(glob: '**.vsix', excludes: vsix[0].path)
-			for(platformVsix in platformVsixes){
-				sh 'vsce publish -p ${TOKEN}' + " --packagePath ${platformVsix.path}"
+		if (publishToMarketPlace.equals('true')) {
```

Comment:
By wrapping this in an if, the prereleases won't publish, since it seems like using `publishToMarketPlace` signals that this is a release build. If instead `publishToMarketPlace` is just meant to signal where to publish, then the popup to click "approve" will appear during prereleases (line 111), blocking the build.

Reply:
Right. When only `publishPreRelease` is set, it should still publish pre-release to all marketplaces.

Reply:
I just added `|| publishPreRelease.equals('true')` to the ifs. That should allow pre-releases to always publish but just selecting that, regardless of marketplace, while allowing us to do proper releases to individual marketplaces. Let me know if I got something wrong.